R/parametricsurvivalanalysis.R

```diff
@@ -17,5 +17,2337 @@
+
+  tempTable <- .sapCreatePredictionTable(
+    options        = options,
+    atTitle        = atTitle,
+    estimateNames  = paste0("par", 1:dataLength, "."),
+    estimateTitles = if (dataLength == 1) estimateTitle else estimateTitles
+  )
+
+  # add remaining information
+  data$at              <- optionsSequence
+  data$subgroup        <- NA
+  data$distribution    <- NA
+  data$model           <- NA
+  data$subgroup[1]     <- attr(fit, "subgroup")
+  data$distribution[1] <- attr(fit, "distribution")
+  data$model[1]        <- attr(fit, "modelTitle")
+
+  if (!is.null(attr(fit, "label")))
+    tempTable$addFootnote(attr(fit, "label"))
+
+  tempTable$setData(data)
+  tempTable$showSpecifiedColumnsOnly <- TRUE
+
+  return(tempTable)
+}
+.sapSurvivalTimeTableFun         <- function(fit, options) {
+
+  tempTable <- .sapCreatePredictionTableWrapper(fit, options, type = "quantile")
+  return(tempTable)
+}
+.sapSurvivalProbabilityTableFun  <- function(fit, options) {
+
+  tempTable <- .sapCreatePredictionTableWrapper(fit, options, type = "survival")
+  return(tempTable)
+}
+.sapHazardTableFun               <- function(fit, options) {
+
+  tempTable <- .sapCreatePredictionTableWrapper(fit, options, type = "hazard")
+  return(tempTable)
+}
+.sapCumHazardTableFun            <- function(fit, options) {
+
+  tempTable <- .sapCreatePredictionTableWrapper(fit, options, type = "cumhaz")
+  return(tempTable)
+}
+.sapRmstTableFun                 <- function(fit, options) {
+
+  tempTable <- .sapCreatePredictionTableWrapper(fit, options, type = "rmst")
+  return(tempTable)
+}
+.sapLifeTimeTableFun             <- function(fit, options) {
+
+  tempTable <- createJaspTable()
+  .sapAddColumnSubgroup(     tempTable, options, output = "coefficientsCovarianceMatrix")
+  .sapAddColumnDistribution( tempTable, options, output = "coefficientsCovarianceMatrix")
+  .sapAddColumnModel(        tempTable, options, output = "coefficientsCovarianceMatrix")
+  tempTable$addColumnInfo(name = "at", title = gettext("Time"), type = "number")
+
+  if (!.saSurvivalReady(options) || jaspBase::isTryError(fit))
+    return(tempTable)
+
+  timeSequence <- .sapOptions2PredictionTime(options, fit)
+  data <- list()
+
+  if (options[["survivalProbabilityTable"]]) {
+    # add dots to column names since cbind merges names with collapse = "."
+    .sapAddColumnsPredictionTable(tempTable, options, estimateTitle = gettext("Survival Probability"), estimateName = "survivalProbability.")
+
+    tempData           <- summary(fit, type = "survival", t = timeSequence, ci = TRUE, cl = options[["confidenceIntervalLevel"]])
+    if (length(tempData) > 1) {
+      tempTable$setError(gettext("Life time tables cannot be merged if there is more than a one prediction from a given model."))
+      return(tempTable)
+    }
+    tempData           <- tempData[[1]][,-1]
+    colnames(tempData) <- c("estimate", "lCi", "uCi")
+    data[["survivalProbability"]] <- tempData
+  }
```

Comment:
In this if statement and each if statement below there is a lot of duplication, can this be avoided with a function?